File: Java/Guess.java
import java.util.concurrent.ThreadLocalRandom;
import java.util.Scanner;

/***********************
 * Guess.java
 *
 * Dustin Michels
 * August 2017
 ************************/


public class Guess {

    public static void main(String []args) {

        // Initialize variables
        int myNum = ThreadLocalRandom.current().nextInt(1, 101); // random int
        int numGuesses = 0;
        int guess;
        Scanner sc  = new Scanner(System.in);

        // Print welcome message
        System.out.println("Hello! I'm thinking of a number between 1 and 100");
        System.out.println("Can you guess it?");

        do {
            System.out.print("Your guess: ");
            while (!sc.hasNextInt()) {
                System.out.println("Oops! Please enter an integer.");
                System.out.print("Your guess: ");
                sc.next();
            }
            guess = sc.nextInt();
            numGuesses++;
            if (guess > myNum) {
                System.out.println("Lower...");
            } else if (guess < myNum) {
                System.out.println("Higher...");
            }
        } while (guess != myNum);

        System.out.printf("%nYeah! It was %d!%n", myNum);
        System.out.printf("You found the answer in %d guesses.", numGuesses);

     }
}

File: Python/guess.py
"""
Number guessing game for Python 3.6+

Generates a (pseudo)random number between 1 and 100, then prompts
the user for guesses until the correct number is found.

Run from the command line by typing: `python3 guess.py`.

Dustin Michels
August 2017
"""


import random


my_num = random.randrange(1, 100)
guess_count = 0

print("\nHello! I'm thinking of a number between 1 and 100. Can you guess it?")

while True:
    guess = input('Your guess: ')

    try:
        guess = int(guess)
    except ValueError:
        print('Oops! Your guess must be an integer between 1 and 100!')

    if guess < my_num:
        print('Higher...')
        guess_count += 1
    elif guess > my_num:
        print('Lower...')
        guess_count += 1
    else:
        print(f'Yeah!! It was {my_num}! Nice job!')
        print(f'You got the right answer in {guess_count} guesses.\n')
